src/CMakeLists.txt

```diff
@@ -1,8 +1,12 @@
 
 # Add source files
 set(SOURCES
-    main.c    
+    TextUtils.c
 )
 
+add_library(tcnn_core STATIC ${SOURCES})
+target_include_directories(tcnn_core PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../include)
+
 # Create executable
-add_executable(tcnn ${SOURCES})
+add_executable(tcnn main.c)
+target_link_libraries(tcnn tcnn_core)
```

Comment:
I separated core tcnn behavior from main so that we could build it into a static library and link it aginst the unittests.